hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/S3EventsHoodieIncrSource.java

```diff
@@ -112,10 +110,15 @@ public S3EventsHoodieIncrSource(
       QueryRunner queryRunner,
       CloudDataFetcher cloudDataFetcher) {
     super(props, sparkContext, sparkSession, schemaProvider);
+
+    if (getBooleanWithAltKeys(props, ENABLE_EXISTS_CHECK)) {
+      sparkSession.conf().set("spark.sql.files.ignoreMissingFiles", "true");
+      sparkSession.conf().set("spark.sql.files.ignoreCorruptFiles", "true");
```

Comment:
Will these configs disrupt when this same sparkSession is used for reading the hudi table ? 
We don't want to skip hudi parquet files if the table is corrupted but rather fail the query.

Reply:
can't we attach this to spark.read calls explicitly. 
```
spark.read.format("").option("spark.sql.files.ignoreCorruptFiles","true")
```
so that we will add these configs only when reading from S3 or GCS bucket. and for any other calls (like hudi reads).

Reply:
See spark docs: https://spark.apache.org/docs/latest/sql-data-sources-generic-options.html#ignore-missing-files `Spark allows you to use the configuration spark.sql.files.ignoreMissingFiles or the data source option ignoreMissingFiles to ignore missing files while reading data from files.`

You need to set `.option("ignoreMissingFiles")` to achieve the behavior.

Reply:
As I stated above, it did not work. Seems similar to issues like[this](https://stackoverflow.com/questions/77195180/ignoremissingfiles-option-not-working-in-certain-scenarios) mentioned online.

Reply:
@bhat-vinay could you show which exact line in Spark or Parquet or Hudi that causes the `FileNotFoundException`?  If it's a Spark issue, we should report that or if it's already fixed in the latest Spark version, we should upgrade Spark.

Reply:
```
org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 0.0 failed 1 times, most recent failure: Lost task 0.0 in stage 0.0 (TID 0) (192.168.0.101 executor driver): java.io.FileNotFoundException: 
File file:/var/folders/9g/815t_9ns1pg2h631kb3792zw0000gn/T/junit5612947518157164794/data1.json does not exist

It is possible the underlying files have been updated. You can explicitly invalidate
the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by
recreating the Dataset/DataFrame involved.
       
	at org.apache.spark.sql.errors.QueryExecutionErrors$.readCurrentFileNotFoundError(QueryExecutionErrors.scala:661)
	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.org$apache$spark$sql$execution$datasources$FileScanRDD$$anon$$readCurrentFile(FileScanRDD.scala:212)
	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:270)
	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:116)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:490)
	at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:364)
	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:890)
	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:890)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:365)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:329)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.run(Task.scala:136)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:548)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1504)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:551)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
```

Reply:
...and https://issues.apache.org/jira/browse/SPARK-38767. Seems the issue is fixed in spark 3.4.0

Reply:
Based on the information, we can only push the `ignoreMissingFile` to Spark for newer Spark releases, which requires using `SparkAdapter` to achieve this and more work in this PR, yet the ROI might not be big enough to justify it.  I think we should deprioritize this now and focus on other inflight PRs.

Reply:
>yet the ROI might not be big enough to justify it

I am not sure of this, but since I have not benchmarked it, we do not know if there are benefits.

>I think we should deprioritize this now and focus on other inflight PRs.

Yes, I already updated the associated jira yesterday and not looking into this for the time being.

Thanks!